base/pkg/libgit2/strarray.jl

```diff
@@ -0,0 +1,25 @@
+# This file is a part of Julia. License is MIT: http://julialang.org/license
+
+function StrArrayStruct{T<:AbstractString}(strs::T...)
+    count = length(strs)
+    strings = convert(Ptr{Ptr{Uint8}}, Libc.malloc(sizeof(Ptr{Uint8}) * count))
+    for i=1:count
+        len = length(strs[i])
+        #in_ptr  = convert(Ptr{Uint8}, bytestring(strs[i]))
+        in_ptr  = pointer(bytestring(strs[i]))
```

Comment:
This looks a lot like the existing Ref{Ptr{UInt8}}(strs) constructor

Sizeof(UInt8) is defined to be 1, so unnecessary to compute it

(From iphone)